File: velocity/src/main/java/gg/grounds/velocity/VelocityDiscoveryConfig.java
package gg.grounds.velocity;

import gg.grounds.discovery.DiscoveryKeys;
import gg.grounds.discovery.ValkeyConfig;

import java.util.Optional;

public record VelocityDiscoveryConfig(
        DiscoveryKeys discoveryKeys,
        ValkeyConfig valkeyConfig
) {
    private static final String ENV_VALKEY_HOST = "VALKEY_HOST";
    private static final String ENV_VALKEY_PORT = "VALKEY_PORT";

    public static VelocityDiscoveryConfig load() {
        DiscoveryKeys keys = new DiscoveryKeys();

        String valkeyHost = Optional.ofNullable(System.getenv(ENV_VALKEY_HOST)).orElse("localhost");
        int valkeyPort = Integer.parseInt(Optional.ofNullable(System.getenv(ENV_VALKEY_PORT)).orElse("6379"));

        ValkeyConfig valkeyConfig = new ValkeyConfig(valkeyHost, valkeyPort);

        return new VelocityDiscoveryConfig(keys, valkeyConfig);
    }
}

File: velocity/src/main/kotlin/gg/grounds/discovery/VelocityDiscoveryConfig.kt
package gg.grounds.discovery

data class VelocityDiscoveryConfig(val valkeyConfig: ValkeyConfig) {
    companion object {
        private const val ENV_VALKEY_HOST = "VALKEY_HOST"
        private const val ENV_VALKEY_PORT = "VALKEY_PORT"

        @JvmStatic
        fun load(): VelocityDiscoveryConfig {
            val valkeyHost = System.getenv(ENV_VALKEY_HOST) ?: "localhost"
            val valkeyPort = (System.getenv(ENV_VALKEY_PORT) ?: "6379").toInt()

            val valkeyConfig = ValkeyConfig(valkeyHost, valkeyPort)

            return VelocityDiscoveryConfig(valkeyConfig)
        }
    }
}
